contracts/cw20-atomic-swap/src/balance.rs

```diff
@@ -1,27 +1,34 @@
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-use cosmwasm_std::Coin;
 use cw20::Cw20Coin;
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum Balance {
-    Native(Vec<Coin>),
+    Native(cw0::Balance),
     Cw20(Cw20Coin),
 }
 
 impl Default for Balance {
     fn default() -> Balance {
-        Balance::Native(vec![])
+        Balance::Native(cw0::Balance(vec![]))
     }
 }
 
 impl Balance {
     pub fn is_empty(&self) -> bool {
         match self {
-            Balance::Native(coins) => coins.is_empty(),
-            Balance::Cw20(_) => false, // FIXME: Reject zero-valued coins
+            Balance::Native(balance) => balance.is_empty(),
```

Comment:
You could make this more robust with something like this:

`! balance.iter().any(|x| x.amount != 0)`

Then we don't need to normalize before checking for empty (actually this logic should be in the `balance.rs` file)

Reply:
Good idea. We don't put `normalize()` inside `is_empty()`, and iterate here. I'll change it.

Reply:
Is this done in CW-escrow as well? The empty check?

It supports a vec of cw20coin.

Maybe there is some unification possible here. Or maybe not. I don't mind is empty as is zero... nothing is there

Reply:
In escrows it's done directly on the `amount` field, i. e. there's no use of `Coin` / `Cw20Coin` there. I looked at it to introduce it, but it wasn't worth it.

The `any`check you propose already works for empty arrays. So, I'm OK with doing that change, and we can keep the `normalize()` outside, as an improvement than can be used by the implementation.